elasticsearch/helpers/vectorstore/_async/_utils.py

```diff
@@ -0,0 +1,51 @@
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from elasticsearch import (
+    AsyncElasticsearch,
+    BadRequestError,
+    ConflictError,
+    NotFoundError,
+)
+
+
+async def model_must_be_deployed(client: AsyncElasticsearch, model_id: str) -> None:
+    try:
+        dummy = {"x": "y"}
+        await client.ml.infer_trained_model(model_id=model_id, docs=[dummy])
+    except NotFoundError as err:
+        raise err
+    except ConflictError as err:
+        raise NotFoundError(
```

Comment:
I assume a 409 in this case means that the model is found, but it is in a state that it cannot perform inferences. If this is correct, why do we want to switch the error to not found, which can be misleading?

Reply:
Let me reproduce this again to understand the failure modes. At the time I was thinking that for the user it's the same, they need to deploy the model (I think because in one case it's not there at all and in the other it's downloaded but not deployed, but let me confirm).

Reply:
Confirmed. I will raise the specific error, agree it's better